File: lab/javascript/pares/pares_entre_dois_numeros.js
function pares_entre_dois_numeros(m, n){
    let bestVal = -1 / 0;  // Number.NEGATIVE_INFINITY; // -Infinity;
    let bestX = 0;
    let bestY = 0;

    for(let x=0; x < m; x++){
        for (let y=0; y < n; y++){
            let E = x * y - x * x + y;

            if (E > bestVal){
                bestVal = E;
                bestX = x;
                bestY = y;
            }
        }
    }

    console.log(bestX, bestY, bestVal);
}


pares_entre_dois_numeros(2, 4);

File: lab/c/pares/pares_entre_dois_numeros.c
#include <stdio.h>
#include <math.h>


void pares_entre_dois_numeros(int m, int n){
    float bestVal = - INFINITY;
    int bestX = 0;
    int bestY = 0;

    for(int x=0; x < m; x++){
        for (int y=0; y < n; y++){
            int E = x * y - x * x + y;

            if (E > bestVal){
                bestVal = E;
                bestX = x;
                bestY = y;
            }
        }
    }

    printf("%d, %d, %.0f\n", bestX, bestY, bestVal);
}


int main(){
    pares_entre_dois_numeros(2, 4);
    return 0;
}
